velox/functions/lib/string/StringImpl.h

```diff
@@ -122,6 +122,25 @@ FOLLY_ALWAYS_INLINE int64_t cappedLength(const T& input, size_t maxLength) {
   }
 }
 
+/// Return a capped bytes(controlled by maxCharacters) of a string.
+/// The returned amount may be greater than maxCharacters if there
+/// are multi-byte characters present in the input string.
+template <typename TString>
+FOLLY_ALWAYS_INLINE size_t
+cappedUnicodeBytes(TString& input, size_t size, size_t maxCharacters) {
+  const char* str = input.data();
+  size_t utf8Position = 0;
+  size_t numCharacters = 0;
+  while (numCharacters < maxCharacters && utf8Position < size) {
```

Comment:
I wonder if it makes sense to change cappedLengthUnicode to return a pair of numbers: length in bytes + length in characters, then re-use it here.